src/Databases/DatabaseOrdinary.cpp

```diff
@@ -274,6 +279,16 @@ void DatabaseOrdinary::alterTable(ContextPtr local_context, const StorageID & ta
     }
 
     commitAlterTable(table_id, table_metadata_tmp_path, table_metadata_path, statement, local_context);
+
+    /// Daisy: starts.
+    StoragePtr table;
+    {
+        std::unique_lock lock{mutex};
```

Comment:
My IDE stuck when i reviewed this code. I felt like we need do

```
{
    std::unique_lock lock{mutex};
    table = getTableUnlocked(table_id.table_name, lock);
    const auto & new_create_query = parseCreateQueryFromAST(ast, database_name, table_id.table_name);
    table->setInMemoryCreateQuery(new_create_query);
}
```

Reply:
I would move the whole function to `DatabaseCommon` like 
```
void DatabaseCommon::setInMemoryCreateQuery(const String & table_name, ast)
{
    std::unique_lock lock{mutex};
    auto table = getTableUnlocked(table_id.table_name, lock);
    const auto & new_create_query = parseCreateQueryFromAST(ast, database_name, table_id.table_name);
    table->setInMemoryCreateQuery(new_create_query);
}
```

Reply:
the `getTableUnlocked` only use here, and the following code is generic
```
const auto & new_create_query = parseCreateQueryFromAST(ast, database_name, table_id.table_name);
table->setInMemoryCreateQuery(new_create_query);
```

Reply:
I know it is used only here, the problem here is we are acquiring a mutex in a deep child class which i really like to avoid. Moving the mutex acquiring logic to a upper parent class makes more sense to me and basically lots of functions are in upper parent class.